Lib/test/test_syntax.py

```diff
@@ -1359,6 +1359,36 @@
    Traceback (most recent call last):
    SyntaxError: cannot have both 'except' and 'except*' on the same 'try'
 
+Better error message for using `except as` with not a name:
+
+   >>> try:
+   ...    pass
+   ... except TypeError as obj.attr:
+   ...    pass
+   Traceback (most recent call last):
+   SyntaxError: cannot use except statement with attribute
```

Comment:
Can we also test that the caret points to the right place?

Reply:
Done! 👍